flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManagerImpl.java

```diff
@@ -496,8 +495,7 @@ public void setFailUnfulfillableRequest(boolean failUnfulfillableRequest) {
 					resourceActions.notifyAllocationFailure(
 						pendingSlotRequest.getJobId(),
 						pendingSlotRequest.getAllocationId(),
-						new ResourceManagerException("Could not fulfill slot request " + pendingSlotRequest.getAllocationId() + ". "
-							+ "Requested resource profile (" + pendingSlotRequest.getResourceProfile() + ") is unfulfillable.")
+						new UnfulfillableSlotRequestException(pendingSlotRequest.getAllocationId(), pendingSlotRequest.getResourceProfile())
```

Comment:
Could we extend `SlotManagerFailUnfulfillableTest` to verify that we throw the correct exception in this method `setFailUnfulfillableRequest` and `internalRequestSlot`?

Reply:
I think the updates have already extended `SlotManagerFailUnfulfillableTest` to verify the exception being thrown. It's in the last commit.

Reply:
Great. Sorry I overlooked this.